### DISCUSSION TITLE
How to automate with Query Rows and Update Row

### DISCUSSION THREAD

**Discussion by gerhardmpl** [Budibase/budibase#4550] [Help] [answered]
I have two table (```information``` and ```service```) and want to update related (multiple) rows of the ```service``` table (linked to the ```information``` table by field _information_name_), when a single row in the ```information``` table is updated.

Automation is set as follows (step 2 can return none, one or many rows):

1. on ```ROW UPDATE``` for a row in ```information``` table
2. ```QUERY ROWS``` from ```service``` table filtering related rows using {{ trigger.row.information_name }}
3. ```UPDATE ROW``` of ```service``` table using {{ trigger.row.information_name }} updating the _confidentiality_ field with {{ trigger.row.confidentiality }}

I can not get step 3 to work in the manual test run and need help on the automation logic / handling (step 3 fails with a _"message": "Invalid inputs"_).

Attached please find my test App with some data.
[automationtest.txt](https://github.com/Budibase/budibase/files/8089277/automationtest.txt)

**Comment by gerhardmpl** [comment]
This is the SQL equivalent of what I'm trying to achieve with step 3 in this automation

```
UPDATE service
SET confidentiality={{ trigger.row.confidentiality }}
WHERE information_name={{ trigger.row.information_name }};
```

**Reply by mjashanks** [reply]
This is not possible at the moment.  The `Update Row` step will only update one row. Your automation is failing because you are not supplying a row id.

In our end-of March release, we are addressing this by adding
1. A `For Each` automation step
2. A `Bulk Update` step

**Comment by gerhardmpl** [comment, marked as answer]
Thanks for explaning, I didn't find that in the documentation or the discussions. Looking forward to the changes you mentioned.
